lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -2306,6 +2308,46 @@ balanceTransaction ctx genChange (ApiT wid) body = do
     toTxOut (TxChange addr amt assets _) =
         TxOut addr (TokenBundle amt assets)
 
+submitTransaction
+    :: forall ctx s k (n :: NetworkDiscriminant).
+        ( ctx ~ ApiLayer s k
+        , HasNetworkLayer IO ctx
+        , IsOwned s k
+        , WalletKey k
+        , HardDerivation k
+        , Typeable s
+        , Typeable n
+        , Bounded (Index (AddressIndexDerivationType k) 'AddressK)
+        )
+    => ctx
+    -> ApiT WalletId
+    -> ApiT W.SealedTx
+    -> Handler ApiTxId
+submitTransaction ctx (ApiT wid) (ApiT sealedTx) = do
+    ttl <- liftIO $ W.getTxExpiry ti Nothing
+    let (Tx txId _ colls inps outs wdrlMap _ _) = tx
+
+    _ <- withWorkerCtx ctx wid liftE liftE $ \wrk -> do
+        (acct, _, _) <- liftHandler $ W.readRewardAccount @_ @s @k @n wrk wid
+        (wdrl, _) <- mkRewardAccountBuilder @_ @s @_ @n ctx wid $
+            if Map.member acct wdrlMap
+            then Just SelfWithdrawal
+            else Nothing
```

Comment:
I really feel uneasy about these conversions... Now that transactions are constructed outside the boundaries of the wallet, there are many invariants / pre-conditions the wallet core functions were holding on that are no longer valid. For example, we use to calculate the _amount_ by diffing inputs+refunds+withdrawals - outputs-deposit, but because we knew that transactions could only ever have one withdrawal or one deposit. With this new API, this is no longer true and may lead to situation where the wallet reports an amount that is completely off. 

The `TransactionContext` and internals of the wallet need to be reworked to work with multi withdrawals and multi deposits and... etc, before we can support such an API I believe.

Reply:
According to https://github.com/input-output-hk/cardano-wallet/pull/2959#discussion_r724909999
I would argue we have here not:
```
inputs+refunds+withdrawals - outputs-deposit
```
but rather 
```
inputs+withdrawals - outputs
```
as deposit paying/returning should be already accounted in inputs/outputs